tachyon/math/finite_fields/quartic_extension_field_unittest.cc

```diff
@@ -0,0 +1,204 @@
+#include <optional>
+
+#include "gtest/gtest.h"
+
+#include "tachyon/math/finite_fields/baby_bear/baby_bear4.h"
+#include "tachyon/math/finite_fields/test/finite_field_test.h"
+
+namespace tachyon::math {
+
+namespace {
+
+using F4 = BabyBear4;
+using F = BabyBear;
+
+class QuaticExtensionFieldTest : public FiniteFieldTest<F4> {};
+
+}  // namespace
+
+TEST_F(QuaticExtensionFieldTest, Zero) {
+  EXPECT_TRUE(F4::Zero().IsZero());
+  EXPECT_FALSE(F4::One().IsZero());
+}
+
+TEST_F(QuaticExtensionFieldTest, One) {
+  EXPECT_TRUE(F4::One().IsOne());
+  EXPECT_FALSE(F4::Zero().IsOne());
+}
+
+TEST_F(QuaticExtensionFieldTest, Random) {
+  bool success = false;
+  F4 r = F4::Random();
+  for (size_t i = 0; i < 100; ++i) {
+    if (r != F4::Random()) {
+      success = true;
+      break;
+    }
+  }
+  EXPECT_TRUE(success);
+}
+
+TEST_F(QuaticExtensionFieldTest, Norm) {
+  constexpr static uint32_t kModulus = BabyBear::Config::kModulus;
+  F4 r = F4::Random();
+  F4 r_to_p = r.Pow(kModulus);
+  F4 r_to_p2 = r_to_p.Pow(kModulus);
+  F4 r_to_p3 = r_to_p2.Pow(kModulus);
+  EXPECT_EQ(r.Norm(), (r * r_to_p * r_to_p2 * r_to_p3).c0());
+}
+
+TEST_F(QuaticExtensionFieldTest, EqualityOperators) {
+  F4 f(F(3), F(4), F(5), F(6));
+  F4 f2(F(4), F(4), F(5), F(6));
+  EXPECT_NE(f, f2);
+  F4 f3(F(4), F(3), F(5), F(6));
+  EXPECT_NE(f2, f3);
+
+  F4 f4(F(3), F(4), F(5), F(7));
+  EXPECT_NE(f, f4);
+
+  F4 f5(F(3), F(4), F(5), F(6));
+  EXPECT_EQ(f, f5);
+}
+
+TEST_F(QuaticExtensionFieldTest, ComparisonOperator) {
+  F4 f(F(3), F(4), F(5), F(6));
+  F4 f2(F(4), F(4), F(5), F(6));
+  EXPECT_TRUE(f < f2);
```

Comment:
And for all following related instances!
```suggestion
  EXPECT_LT(f, f2);
```